src/votekit/ballot_generator.py

```diff
@@ -509,84 +515,110 @@ class AlternatingCrossover(BallotGenerator):
-                    pref_for_bloc = [pref_interval_dict[cand] for cand in bloc_cands]
-                    # convert to probability distribution
-                    pref_for_bloc = [p / sum(pref_for_bloc) for p in pref_for_bloc]
-
-                    bloc_cands = list(
-                        np.random.choice(
-                            bloc_cands,
-                            p=pref_for_bloc,
-                            size=len(bloc_cands),
-                            replace=False,
-                        )
+            opposing_slate = list(set(self.bloc_voter_prop.keys()).difference([bloc]))[0]
+            opposing_cands = self.slate_to_candidates[opposing_slate]
+            bloc_cands = self.slate_to_candidates[bloc]
+
+            pref_for_opposing = [
+                    pref_interval_dict[cand] for cand in opposing_cands
+                ]
+            # convert to probability distribution
+            pref_for_opposing = [
+                p / sum(pref_for_opposing) for p in pref_for_opposing
+            ]
+
+            pref_for_bloc = [pref_interval_dict[cand] for cand in bloc_cands]
+            # convert to probability distribution
+            pref_for_bloc = [p / sum(pref_for_bloc) for p in pref_for_bloc]
+
+            for _ in range(num_cross_ballots):
+                # selects order of bloc, opposing candidates by PL
+                bloc_cands = list(
+                    np.random.choice(
+                        bloc_cands,
+                        p=pref_for_bloc,
+                        size=len(bloc_cands),
+                        replace=False,
                     )
-                    opposing_cands = list(
-                        np.random.choice(
-                            opposing_cands,
-                            size=len(opposing_cands),
-                            p=pref_for_opposing,
-                            replace=False,
-                        )
+                )
+                opposing_cands = list(
+                    np.random.choice(
+                        opposing_cands,
+                        size=len(opposing_cands),
+                        p=pref_for_opposing,
+                        replace=False,
                     )
+                )
 
-                    # alternate the bloc and opposing bloc candidates to create crossover ballots
-                    if bloc != opposing_slate:  # alternate
-                        ballot = [
-                            item
-                            for pair in zip(opposing_cands, bloc_cands)
-                            for item in pair
-                            if item is not None
-                        ]
-
-                    # check that ballot_length is shorter than total number of cands
-                    ballot_pool.append(ballot)
+                # alternate the bloc and opposing bloc candidates to create crossover ballots
+                ranking = [{cand} for pair in zip(opposing_cands, bloc_cands)\
```

Comment:
Per my above comment, using an `if` statement here to switch how we create the ranking based on what iteration (or number of ballots generated) we are on in the for loop.